File: p463.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    int islandPerimeter(vector<vector<int>>& grid) {
        int answer = 0;
        int n = grid.size();
        int m = grid[0].size();
        for(int i = 0; i < n; ++i) {
            for(int j = 0; j < m; ++j) {
                if(grid[i][j] == 1) {
                    if(i == 0 || grid[i-1][j] == 0)
                        ++answer;
                    if(i == n-1 || grid[i+1][j] == 0)
                        ++answer;
                    if(j == 0 || grid[i][j-1] == 0)
                        ++answer;
                    if(j == m-1 || grid[i][j+1] == 0)
                        ++answer;
                }
            }
        }
        return answer;
    }
};

File: p463.py
class Solution:
    def islandPerimeter(self, grid: List[List[int]]) -> int:
        m, n = len(grid), len(grid[0])
        ans = 0

        def count(i, j):
            ret = 0
            if i-1 == -1 or grid[i-1][j] == 0:
                ret += 1
            if i+1 == m or grid[i+1][j] == 0:
                ret += 1
            if j-1 == -1 or grid[i][j-1] == 0:
                ret += 1
            if j+1 == n or grid[i][j+1] == 0:
                ret += 1
            return ret

        for i in range(m):
            for j in range(n):
                if grid[i][j]:
                    ans += count(i, j)

        return ans
